src/main/java/org/apache/maven/plugins/jar/AbstractJarMojo.java

```diff
@@ -262,7 +294,7 @@ public File createArchive() throws MojoExecutionException {
                     getLog().warn("JAR will be empty - no content was marked for inclusion!");
                 }
             } else {
-                archiver.getArchiver().addDirectory(contentDirectory, getIncludes(), getExcludes());
+                archiver.getArchiver().addFileSet(getFileSet(contentDirectory));
```

Comment:
Call me stupid, but isn't this exactly what `jarContentFileSet` is about? What am I missing here?

Reply:
`jarContentFileSet` is of type `org.apache.maven.shared.model.fileset.FileSet`, while `plexus-archiver` accepts `org.codehaus.plexus.archiver.FileSet`.

Unfortunately there is no way of setting the `usingDefaultExcludes` flag in a flowable way so the method is a bit longer that it could have been.
PS. `addDirectory` method of `org.codehaus.plexus.archiver.AbstractArchiver` has been deprecated in the latest version of  the archiver so moving away from it makes sense.

Reply:
I need to look at it, but have never applied any real fixes to this plugin. It won't happen before next month. Maybe someone with more knowledge can intervene.

Reply:
I would like to use only `org.codehaus.plexus.archiver.FileSet`, there is a method: `getIncludes` so we can drop a using of `rg.apache.maven.shared.model.fileset.FileSet` at all.
Also dependency to `file-management` can be removed.

Reply:
Ok, it looks a more complicated `org.codehaus.plexus.archiver.FileSet` is only a configuration class which is used in plexus archiver ....